### DISCUSSION TITLE
Output Width Height and DPI of a PNG to a CSV file

### DISCUSSION THREAD

**Discussion by dldsales** [ImageMagick/ImageMagick#7748] [Help]
### ImageMagick version

7.1.1.37

### Operating system, version and so on

WINDOWS10

### Description

Hello I am trying to Output Width Height and DPI of a PNG to a CSV file. I am currently using the following:

identify -format "%f,%w,%h\n" *.png > pnglist.csv

How can I get the DPI of the PNG? some of my images are 72dpi and MOST are 300dpi. I need to know the actual print width of the image. So when I get to excel I calcultate widthXheightXdpi to get inches. I currently use a default 300 dpi when making the calculation, however now and then an image creeps into the directory that is not 300dpi!

Is there another way to get the Width and Height of the image in INCHES?

Thank You

**Comment by fmw42** [comment]
See %x and %y at https://imagemagick.org/script/escape.php

**Reply by dldsales** [reply]
Thank you for your reply. I have used %x and %Y before. I am looking for a third parameter to output and that is the DPI of the image. 

This command outputs
identify -format "%f,%w,%h\n" *.png > pnglistwh.csv

OUTPUT
CIAA_logo_3.5 wide-1.png	1050	649
Gift Bear-HEATTRANSFER-2.png	4500	4500
Jermale Hall Back-22.png	3600	2525
Jermale Hall Front-22.png	864	728
Northwest Valley FINAL-_2-10.png	1201	876

I am trying to achieve this OUTPUT, where the last field is DPI
CIAA_logo_3.5 wide-1.png	1050	649	300
Gift Bear-HEATTRANSFER-2.png	4500	4500	300
Jermale Hall Back-22.png	3600	2525	300
Jermale Hall Front-22.png	864	728	72
Northwest Valley FINAL-_2-10.png	1201	876	300

Is this possible?

Thank You

**Comment by snibgo** [comment]
I suggest:
```
-units pixelsperinch -format %x\n 
```
Strictly speaking, %x is the horizontal resolution. The vertical resolution %y may be different. But I suppose you want to ignore that.

EDIT: If you want the width and height in inches:
```
-units pixelsperinch -format "%[fx:w/%[x]] %[fx:h/%[y]]\n"
```
EDIT2: That last one can be re-written more naturally as:
```
-units pixelsperinch -format "%[fx:w/resolution.x] %[fx:h/resolution.y]\n"
```
